build_web_compilers/lib/src/web_entrypoint_builder.dart

```diff
@@ -26,16 +28,33 @@ const mergedMetadataExtension = '.dart.ddc_merged_metadata';
 /// Which compiler to use when compiling web entrypoints.
 enum WebCompiler {
   // ignore: constant_identifier_names
-  Dart2Js('dart2js'),
+  Dart2Js('dart2js', jsEntrypointExtension, '.dart2js.js'),
   // ignore: constant_identifier_names
-  DartDevc('dartdevc'),
+  DartDevc('dartdevc', jsEntrypointExtension, '.ddc.js'),
   // ignore: constant_identifier_names
-  Dart2Wasm('dart2wasm');
+  Dart2Wasm('dart2wasm', moduleJsExtension, moduleJsExtension);
 
   /// The name of this compiler used when identifying it in builder options.
   final String optionName;
 
-  const WebCompiler(this.optionName);
+  /// The [EnabledEntrypointCompiler.extension] to use when this compiler is the
+  /// only compiler enabled.
+  /// This defaults to `.dart.js` for compilers targeting JavaScript and `.mjs`
+  /// for the module file emitted by `dart2wasm`.
+  final String entrypointExtensionWhenOnlyCompiler;
+
+  /// The [EnabledEntrypointCompiler.extension] to use by default when multiple
+  /// compilers are enabled.
+  ///
+  /// This can't be `.dart.js` for multiple web compilers because we'd then run
+  /// into duplicate outputs being generated.
+  final String entrypointExtension;
+
+  const WebCompiler(
+    this.optionName,
+    this.entrypointExtensionWhenOnlyCompiler,
```

Comment:
I would make these named, it isn't clear looking at a constructor invocation which is which, especially since they are the same type